client/app/src/Form/ClientType.php

```diff
@@ -30,19 +31,25 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 ->add('address5', FormTypes\TextType::class)
                 ->add('postcode', FormTypes\TextType::class)
                 ->add('country', FormTypes\CountryType::class, [
-                      'preferred_choices' => ['GB'],
-                      'placeholder' => 'country.defaultOption',
+                    'preferred_choices' => ['GB'],
+                    'placeholder' => 'country.defaultOption',
                 ])
                 ->add('phone', FormTypes\TextType::class)
                 ->add('id', FormTypes\HiddenType::class)
                 ->add('save', FormTypes\SubmitType::class);
 
         // strip tags to prevent XSS as the name is often displayed around mixed with translation with the twig "raw" filter
+        // only allow user to enter a four digit year
         $builder->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $event) {
             $data = $event->getData();
             $data['firstname'] = strip_tags($data['firstname']);
             $data['lastname'] = strip_tags($data['lastname']);
             $event->setData($data);
+
+            if (!preg_match('/^\d{4}$/', $data['courtDate']['year'])) {
```

Comment:
We should move this to some form of form validation rather than logic. Also should complete a check that's not in the future. Maybe also too far in the past, not sure there is any valid cases in the 1800's